### PR TITLE
added akita ng fire hook when logging out

### LINKED ISSUE #320: Clean /reset all stores on user logout

### PR DESCRIPTION
**Pull request by Maxstgt**
close #320

### CODE REVIEW

File: apps/catalog/catalog/src/app/marketplace/layout/layout.component.ts [hunk 1]
```diff
@@ -44,10 +44,4 @@ export class LayoutComponent implements OnInit, AfterViewInit, OnDestroy {
   ngOnDestroy() {
     this.sub.unsubscribe();
   }
-
-  public async logout() {
-    await this.authService.signOut();
-    // TODO: issue#879, navigate with router
-    window.location.reload();
-  }
```
Comment (Maxstgt) [inline apps/catalog/catalog/src/app/marketplace/layout/layout.component.ts]: function was never called

File: libs/auth/src/lib/+state/auth.service.ts [hunk 2]
```diff
@@ -38,6 +38,16 @@ export class AuthService extends FireAuthService<AuthState> {
...
   }
 
+  /**
+   * @description function that gets triggered when
+   * AuthService.signOut is called
+   */
+  public onSignout() {
+    localStorage.clear();
+    indexedDB.deleteDatabase('firebase-installations-database');
+    indexedDB.deleteDatabase('firebaseLocalStorageDb');
```
Comment (phpgeekfr) [inline libs/auth/src/lib/+state/auth.service.ts]: Can you add some comments so other can understand what this function is doing?

Comment (phpgeekfr) [reply libs/auth/src/lib/+state/auth.service.ts]: Once logged out there is no remaining user data on browser side ?

Comment (Maxstgt) [reply libs/auth/src/lib/+state/auth.service.ts]: yes that is how I understood the requirements of the issue. Like the example you said that maybe the user is on a shared desktop and we dont want to have any data remaining.

**Pull request merged**